src/watson.ts

```diff
@@ -0,0 +1,824 @@
+// https://tailordev.github.io/Watson/
+const listProjects: Fig.Generator = {
+  script: `watson projects`,
+  postProcess: (output) => {
+    return output.split("\n").map((project) => ({
+      name: project,
+      icon: "🗂",
+    }));
+  },
+};
+
+const listTags: Fig.Generator = {
+  script: `watson tags`,
+  postProcess: (output) => {
+    return output.split("\n").map((tag) => ({
+      name: tag,
+      icon: "🏷",
+    }));
+  },
+};
+
+const listFrames: Fig.Generator = {
+  script: `watson log --json --reverse`,
+  postProcess: (output) => {
+    return JSON.parse(output).map((frame) => ({
+      name: frame.id.substring(0, 7),
+      icon: "⏲",
+      displayName: `${frame.id.substring(0, 7)} - ${frame.project} - ${
+        frame.start
+      }`,
+    }));
+  },
+};
+
+const completionSpec: Fig.Spec = {
+  name: "watson",
+  description: "A wonderful CLI to track your time",
+  subcommands: [
+    {
+      name: "add",
+      description:
+        "Add time to a project with tag(s) that was not tracked live",
+      args: {
+        generators: [listProjects, listTags],
+      },
+      options: [
+        {
+          name: ["--from", "-f"],
+          description: "Date and time of start of tracked activity",
+          args: {
+            name: "DATETIME",
+            description: "The start time. Ex: 2022-03-04 12:00:00",
+          },
+        },
+        {
+          name: ["--to", "-t"],
+          description: "Date and time of end of tracked activity",
+          args: {
+            name: "DATETIME",
+            description: "The start time. Ex: 2022-03-04 12:00:00",
+          },
+        },
+        {
+          name: ["--confirm-new-project", "-c"],
+          description: "Confirm creation of new tag",
+        },
+      ],
+    },
+    {
+      name: "aggregate",
+      description:
+        "Display a report of the time spent on each project aggregated by day",
+      options: [
+        {
+          name: ["--current", "-c"],
+          description: "Include currently running frame in report",
+          exclusiveOn: ["--no-current", "-C"],
```

Comment:
nitpick: it is unnecessary to specify all the flags for each exclusive option, one is sufficient:
e.g.
```suggestion
          exclusiveOn: ["-C"],
```